cfme/cloud/instance/image.py

```diff
@@ -1,40 +1,127 @@
+    VMEntities, EditView, SetOwnershipView, ManagePoliciesView, PolicySimulationView, EditTagsView,
+    BasicProvisionFormView)
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigate_to, CFMENavigateStep, navigator
-from utils.log import logger
+from . import CloudInstanceView, InstanceAccordion
 
-# TODO get rid of this garbage with widgets
-cfg_btn = partial(tb.select, 'Configuration')
-life_btn = partial(tb.select, 'Lifecycle')
-pol_btn = partial(tb.select, 'Policy')
 
-list_table = PagedTable(table_locator="//div[@id='list_grid']//table")
+class ImageToolbar(View):
+    """
+    Toolbar view for image collection
+    """
+    reload = Button(title='Reload current display')
+    configuration = Dropdown('Configuration')
+    lifecycle = Dropdown('Lifecycle')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
 
-match_page = partial(match_location, controller='vm_cloud', title='Instances')
+    view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-def details_page_check(name, provider):
-    title_match = match_page(summary='Image "{}"'.format(name))
-    if title_match:
-        # Also check provider
-        # Limits scope of testing to images that aren't orphaned or archived, but if we don't do
-        # this and multiple providers are present we might have multiple images with the same name
-        try:
-            prov_match = InfoBlock.text('Relationships', 'Cloud Provider') == provider
-            return title_match and prov_match
-        except BlockTypeUnknown:
-            # Default to false since we can't identify which provider the image belongs to
-            return False
+class ImageDetailsToolbar(View):
+    """
+        Toolbar view for image collection
+        """
+    reload = Button(title='Reload current display')
+    configuration = Dropdown('Configuration')
+    lifecycle = Dropdown('Lifecycle')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
+
+
+class ImageDetailsEntities(View):
+    title = Text('//div[@id="main-content"]//h1//span[@id="explorer_title_text"]')
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
+                          '/div[@id="flash_text_div" or contains(@class, "flash_text_div")]')
+    properties = SummaryTable(title='Properties')
+    lifecycle = SummaryTable(title='Lifecycle')
+    relationships = SummaryTable(title='Relationships')
+    compliance = SummaryTable(title='Compliance')
+    power_management = SummaryTable(title='Power Management')
+    security = SummaryTable(title='Security')
+    configuration = SummaryTable(title='Configuration')
+    smart_management = SummaryTable(title='Smart Management')
+
+
+class ImageAllView(CloudInstanceView):
+    """View for the Image collection"""
+    toolbar = View.nested(ImageToolbar)
+    sidebar = View.nested(InstanceAccordion)
+    including_entities = View.include(VMEntities, use_parent=True)
```

Comment:
woud you mind using BaseEntities as we discussed before ?

Reply:
done and done, forgot I had used it here too.